replica/table.cc

```diff
@@ -3420,6 +3420,8 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
                                             tid, _schema->ks_name(), _schema->cf_name()));
         }
 
+        co_await db.clear_inactive_reads_for_tablet(_schema->id(), cg_ptr->token_range());
+
```

Comment:
Q: what happens if a read is active while this is running, and becomes inactive after this completes running?

I think it's safe. While the read is active we'll hold effective_replication_map_ptr, so no transition can take place.

Reply:
Yes, tablet cleanup should take care of fencing active reads. Also note that I took great care to collect all inactive reads without preemption, so no other fiber can interfere with that part.